### PR TITLE
Code Clean up: Use Pattern Matching

### PR DESCRIPTION
**Pull request by BaronGreenback**
Part 1

### CODE REVIEW

File: MediaBrowser.Controller/Entities/TV/Series.cs [hunk 1]
```diff
@@ -318,20 +318,13 @@ public async Task RefreshAllMetadata(MetadataRefreshOptions refreshOptions, IPro
...
-                {
-                    skipItem = true;
-                }
+                bool skipItem = item is Episode episode
+                                && refreshOptions.MetadataRefreshMode != MetadataRefreshMode.FullRefresh
+                                && !refreshOptions.ReplaceAllMetadata
+                                && episode.IsMissingEpisode
+                                && episode.LocationType == LocationType.Virtual
+                                && episode.PremiereDate.HasValue
+                                && (DateTime.UtcNow - episode.PremiereDate.Value).TotalDays > 30;
```
Comment (dkanada) [inline MediaBrowser.Controller/Entities/TV/Series.cs:327]: this should be an if statement imo

File: MediaBrowser.Providers/MediaInfo/AudioImageProvider.cs [hunk 2]
```diff
@@ -137,9 +137,7 @@ public bool Supports(BaseItem item)
                 return false;
             }
 
-            var audio = item as Audio;
-
-            return audio != null;
+            return item is Audio audio;
```
Comment (crobibero) [inline MediaBrowser.Providers/MediaInfo/AudioImageProvider.cs:140]:
```suggestion
            return item is Audio;
```

Don't need to store this

File: MediaBrowser.Controller/Entities/TV/Series.cs (hunk 1 above)
Comment (barronpm) [inline MediaBrowser.Controller/Entities/TV/Series.cs:327]:
```suggestion
                bool skipItem = item is Episode episode
                    && refreshOptions.MetadataRefreshMode != MetadataRefreshMode.FullRefresh
                    && !refreshOptions.ReplaceAllMetadata
                    && episode.IsMissingEpisode
                    && episode.LocationType == LocationType.Virtual
                    && episode.PremiereDate.HasValue
                    && (DateTime.UtcNow - episode.PremiereDate.Value).TotalDays > 30;
```

Comment (barronpm) [reply MediaBrowser.Controller/Entities/TV/Series.cs:327]: I disagree, since the only thing the if statement would do is assign the variable anyways

**Review by barronpm — APPROVED** [review]

File: MediaBrowser.Controller/Entities/TV/Series.cs (hunk 1 above)
Comment (Keridos) [reply MediaBrowser.Controller/Entities/TV/Series.cs]: The logical statement looks almost identical to the if statement. And I personally find this version cleaner and easier to read.

**Review by Ullmie02 — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**